lib/auth/auth_with_roles.go

```diff
@@ -3970,10 +3970,32 @@ func (a *ServerWithRoles) DeleteNamespace(name string) error {
 
 // GetRoles returns a list of roles
 func (a *ServerWithRoles) GetRoles(ctx context.Context) ([]types.Role, error) {
-	if err := a.action(apidefaults.Namespace, types.KindRole, types.VerbList, types.VerbRead); err != nil {
-		return nil, trace.Wrap(err)
+	authErr := a.action(apidefaults.Namespace, types.KindRole, types.VerbList, types.VerbRead)
+	if authErr == nil {
```

Comment:
The return at this point seems incorrect to me.  Don't we need to check the `where` conditions for this condition as well?

Reply:
Actually no, in this case the user just has general, no-where-clause RBAC access, so we can safely return here.

Reply:
Thank you